### ISSUE TITLE
2.4.* - remove "as strong as verification server performance allow" part from requirements

### ISSUE THREAD

**Issue by elarlang** [OWASP/ASVS#1510]
As an example with `bcrypt`, in ASVS v4.0.3 we had:
_V2.4.4 Verify that if bcrypt is used, the work factor SHOULD be as large as verification server performance will allow, with a minimum of 10._

I suggested to remove "_as large as verification server performance will allow, with a_" part (https://github.com/OWASP/ASVS/issues/1159#issuecomment-1175297625)

and we have now:
_Verify that if bcrypt is used, the work factor is a minimum of 10 and password size is limited to 72-bytes due to bcrypt's input limit._

I think we should get rid of this part from other similar requirements as well, as it's not possible to verify that in practice:

| # | Description | L1 | L2 | L3 | CWE | [NIST &sect;](https://pages.nist.gov/800-63-3/sp800-63b.html) |
| :---: | :--- | :---: | :---: | :---: | :---: | :---: |
| **2.4.3** | [MODIFIED] Verify that if PBKDF2 is used, the iteration count should **be as large as verification server performance will allow, with** a minimum of 720,000 iterations with PBKDF2-HMAC-SHA1, a minimum of 310,000 iterations using PBKDF2-HMAC-SHA256, or with a minimum of 120,000 iterations with PBKDF2-HMAC-SHA512. ([C6](https://owasp.org/www-project-proactive-controls/#div-numbering)) | | ✓ | ✓ | 916 | 5.1.1.2 |
| **2.4.4** | [MODIFIED] Verify that if bcrypt is used, the work factor is a minimum of 10 and password size is limited to 72-bytes due to bcrypt's input limit. ([C6](https://owasp.org/www-project-proactive-controls/#div-numbering)) | | ✓ | ✓ | 916 | 5.1.1.2 |
| **2.4.5** | [DELETED] | | | | | |
| **2.4.6** | [ADDED] Verify that if argon2id is used, the configuration should **be as strong as verification server performance will allow, with** a minimum configuration of 15 MiB of memory, an iteration count of 2, and 1 degree of parallelism. ([C6](https://owasp.org/www-project-proactive-controls/#div-numbering)) | | ✓ | ✓ | 916 | 5.1.1.2 |
| **2.4.7** | [ADDED] Verify that if scrypt is used, the configuration should be **as strong as verification server performance will allow, with** a minimum CPU/memory cost parameter of (2^16), a minimum block size of 8 (1024 bytes), and a parallelization parameter of 1. ([C6](https://owasp.org/www-project-proactive-controls/#div-numbering)) | | ✓ | ✓ | 916 | 5.1.1.2 |

**Comment by tghosth** [comment]
@jmanico do you disagree?

@set-reminder 2 weeks @tghosth to merge if no response/dissent

**Comment by suyash5053** [comment]
@elarlang You basically want to delete the whole 2.4.4?

**Comment by elarlang** [comment]
> @elarlang You basically want to delete the whole 2.4.4?

No, I suggest to delete phrase like proposed in initial issue description:

> I suggested to remove "**as large as verification server performance will allow, with a**" part
